### PR TITLE
CXP-1240: Be able to get families from catalogs context

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/src/Infrastructure/Persistence/SearchFamilyQuery.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+use Akeneo\Pim\Structure\Component\Model\FamilyInterface;
+use Akeneo\Tool\Component\StorageUtils\Repository\SearchableRepositoryInterface;
+
+/**
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class SearchFamilyQuery
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Persistence/SearchFamilyQuery.php:15]: no interface ?

Comment (willy-ahva) [reply components/catalogs/back/src/Infrastructure/Persistence/SearchFamilyQuery.php:15]: I asked myself, but I think there is no need here. It's internal to the infra.

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetFamiliesActionTest.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        );
+
+        $response = $client->getResponse();
+        Assert::assertEquals(200, $response->getStatusCode());
+
+        $families = \json_decode($response->getContent(), true);
+        Assert::assertCount(3, $families);
+        Assert::assertArrayHasKey('code', $families[0]);
+        Assert::assertArrayHasKey('label', $families[0]);
+    }
```
Comment (masacc) [inline components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetFamiliesActionTest.php:112]: This test seems to be a remaining copy paste, you can remove it

**Review by masacc — APPROVED** [review]

File: components/catalogs/back/tests/Unit/Infrastructure/Controller/Internal/GetFamiliesActionTest.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Catalogs\Test\Unit\Infrastructure\Controller\Internal;
```
Comment (RiffVentura) [inline components/catalogs/back/tests/Unit/Infrastructure/Controller/Internal/GetFamiliesActionTest.php:5]: I thought we use only integration tests for controllers

**Review by RiffVentura — APPROVED** [review]

**Review by tseho — APPROVED** [review]

**Pull request merged**